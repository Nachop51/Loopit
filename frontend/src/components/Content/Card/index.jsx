import "./Card.css";
import PropTypes from "prop-types";

const Card = ({ label, text, children }) => {
  return (
    <div className="feature-box">
      {children}
      <h3 className="feature-tertiary">{label}</h3>
      <p className="feature-p">{text}</p>
    </div>
  );
};

Card.propTypes = {
  label: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
};

export default Card;
