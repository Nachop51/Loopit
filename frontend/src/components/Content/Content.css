.intro {
  @apply grid py-12 px-4 grid-cols-1 gap-4;
  @apply sm:px-12;
  @apply md:place-items-center md:py-40;
  @apply lg:gap-x-8 lg:gap-y-4 lg:px-14 lg:py-52;
  @apply xl:grid-cols-2;
  background: linear-gradient(90deg, #09203f 0%, #537895 100%), #ffffff;
}

.intro-text {
  @apply flex flex-col justify-center mt-14;
  @apply sm:mt-16;
  @apply md:mt-16;
  @apply lg:justify-start;
}

.intro-title {
  @apply px-2 text-3xl;
  @apply md:mb-4 md:text-5xl;
  @apply lg:text-6xl;
  text-shadow: 0 4px 8px rgba(0, 0, 0, 0.12), 0 2px 4px rgba(0, 0, 0, 0.08);
  /* background: linear-gradient(90deg, #09203f 0%, #537895 100%); */ /* ^ Light mode */
  background: linear-gradient(270deg, #5a6a80 0%, #e7f5ff 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  color: transparent;
  background-clip: text;
}

.intro-p {
  @apply px-2 py-4 sm:text-justify text-gray-400 mb-5;
}

.intro-images {
  @apply flex flex-col-reverse justify-center items-center;
  @apply sm:flex sm:flex-row sm:justify-center sm:items-center;
}

.intro-images-laptop {
  @apply mb-4;
  @apply w-full sm:w-2/3  md:w-2/3 lg:w-2/3; /* * xl: for extra large screens */
}

.intro-images-phone {
  @apply w-3/5 sm:w-1/3 md:w-1/5 lg:w-1/5;
}

.go {
  @apply h-11 mt-10 !mb-20 mx-auto;
  @apply sm:mb-20;
  @apply md:pb-10;
}

/* * FEATURES */
.features {
  background-color: #e7f5ff;
  @apply pt-8 pb-24;
}

.row {
  @apply grid grid-cols-1 place-items-center gap-8 mt-16;
  @apply xl:grid-cols-3 xl:px-48;
}
