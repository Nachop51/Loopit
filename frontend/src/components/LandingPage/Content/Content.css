.main {
  @apply sm:mx-12;
  @apply lg:mx-14;
}

.intro {
  @apply mt-24 mx-4;
  @apply grid;
  @apply md:place-items-center;
  @apply lg:gap-x-8;
  @apply xl:grid-cols-2;
  /* background: linear-gradient(90deg, #09203f 0%, #537895 100%), #ffffff; */
  min-height: calc(100vh - 6rem);
  border-radius: 1.2rem 1.2rem 0 0;
}

.intro-text {
  @apply flex flex-col;
  @apply lg:justify-start;
}

.intro-text__title {
  @apply text-3xl text-transparent text-center my-6 bg-slate-50;
  @apply md:text-5xl md:my-10;
  @apply lg:text-6xl lg:my-12;
  @apply xl:text-left xl:text-8xl;
  background-color: #f6feff;
  -webkit-text-fill-color: transparent;
  -webkit-background-clip: text;
  background-clip: text;
  text-shadow: 2px 4px 4px rgb(0, 0, 0, 0.1);
}

.intro-text__p {
  @apply pt-6 sm:text-justify text-gray-300 text-xl mb-5 font-medium;
}

.intro-text__button {
  @apply mx-auto mt-14;
}

.intro-images {
  @apply my-auto pb-14 flex flex-col-reverse justify-center items-center;
  @apply sm:flex sm:flex-row sm:justify-center sm:items-center;
}

.intro-images__laptop {
  @apply mb-4;
  @apply w-full sm:w-2/3  md:w-2/3 lg:w-2/3; /* * xl: for extra large screens */
}

.intro-images__phone {
  @apply w-3/5 sm:w-1/3 md:w-1/5 lg:w-1/5;
}

/* * FEATURES */
.features {
  /* background-image: linear-gradient(170deg, #d0eeff, #f6feff 50%, #d0eeff); */
  @apply pt-8 pb-24 mb-10;
  margin-bottom: 2.5rem;
  padding-top: 2rem;
  padding-bottom: 6rem;
  border-radius: 0 0 1.2rem 1.2rem;
  /* background: radial-gradient(#ffffff45, #ffffff00); */

  border-radius: 20px;
}

.intro-text__button {
  @apply xl:ml-0;
}

.row {
  @apply grid grid-cols-1 place-items-center gap-8 mt-16;
  @apply xl:grid-cols-3 xl:px-48;
}

.intro-text__button.btn.btn-animation {
  @apply bg-slate-900 text-white mb-14 w-2/5;
}
