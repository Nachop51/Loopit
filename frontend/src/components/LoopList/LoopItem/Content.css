pre {
  border-radius: 3px;
  max-width: 80ch;
}
