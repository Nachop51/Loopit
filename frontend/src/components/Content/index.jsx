import Laptop from "../../assets/laptop.png";
import Phone from "../../assets/phone.png";
import { GrShareOption } from "react-icons/gr";
import { FiCopy, FiDownload } from "react-icons/fi";
import Card from "./Card";
import TitleAnimation from "./TitleAnimation";
import "./Content.css";

const Content = () => {
  return (
    <main>
      <section className="intro">
        <div className="intro-text">
          <h1 className="intro-title">Learn from the best solutions</h1>
          <p className="intro-p">
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Odio
            laboriosam nulla molestias soluta amet ab natus aut corrupti odit
            quas doloribus dignissimos dicta similique consequuntur neque
            molestiae, totam cum nobis. Lorem ipsum dolor sit amet consectetur,
            adipisicing elit. Voluptate maxime obcaecati iusto quae debitis
            suscipit at deleniti architecto labore dicta fugiat, atque aliquam
            id illum. Praesentium reprehenderit dolores asperiores officiis.
          </p>
          <button className="button go btn">Go to app</button>
        </div>
        <div className="intro-images">
          <img className="intro-images-phone" src={Phone} alt="Phone" />
          <img className="intro-images-laptop" src={Laptop} alt="Laptop" />
        </div>
      </section>
      <section className="features">
        <TitleAnimation title="Awesome features!" />
        <div className="row">
          <Card
            label="Share your code!"
            text="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Molestiae omnis ipsa."
          >
            <GrShareOption className="feature-icon" />
          </Card>
          <Card
            label="Copy"
            text="dolore fuga enim sed explicabo iusto nostrum cumque temporibus itaque nesciunt."
          >
            <FiCopy className="feature-icon" />
          </Card>
          <Card
            label="Download the code!"
            text="voluptate dolores ea atque ex. Ab, aut labore?."
          >
            <FiDownload className="feature-icon" />
          </Card>
        </div>
      </section>
    </main>
  );
};

export default Content;
