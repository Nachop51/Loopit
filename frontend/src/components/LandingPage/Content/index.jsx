import { useNavigate } from "react-router-dom";
import { FiCopy, FiDownload, FiShare2 } from "react-icons/fi";

// import Laptop from "../../../assets/sombra-negra-mockup-blanco.png";
import Laptop from "../../../assets/sombra-blanca-mockup-dark.png";

// import Phone from "../../../assets/mockup-mobile.png";
import Card from "./Card";
import TitleAnimation from "./TitleAnimation";
import "./Content.css";

const Content = ({ userStatus, openModal }) => {
  const navigate = useNavigate();

  const goApp = (link) => {
    navigate(link);
  };

  return (
    <main className="main">
      <section className="intro">
        <article className="intro-text">
          <h1 className="intro-text__title">
            Discover new <br /> solutions <u>daily.</u>
          </h1>
          <p className="intro-text__p">
            Loopit is a social media platform for programmers to share their
            code with the world. Create a profile, upload your code, and share
            it with your friends and followers.
          </p>
          <div className="intro-text__button-container">
            <button
              className="intro-text__button btn btn-animation btn-primary"
              onClick={() => {
                if (userStatus) goApp("/l");
                else openModal();
              }}
            >
              Go to app
            </button>
            <button
              className="intro-text__button btn btn-animation"
              onClick={() => goApp("/public")}
            >
              Preview
            </button>
          </div>
        </article>
        <article className="intro-images">
          {/* <img className="intro-images__phone" src={Phone} alt="Phone" /> */}
          <img className="intro-images__laptop" src={Laptop} alt="Laptop" />
        </article>
      </section>
      <section className="features">
        <TitleAnimation title="Awesome features!" width={17} />
        <div className="row">
          <Card
            label="Follow other programmers"
            text="Share your code with the world and get feedback from other programmers."
          >
            <FiShare2 className="feature-icon" />
          </Card>
          <Card
            label="Easily search code"
            text="Instantly copy code snippets to your clipboard."
          >
            <FiCopy className="feature-icon" />
          </Card>
          <Card
            label="Save your favorite solutions"
            text="Download the code to your computer in just one click!"
          >
            <FiDownload className="feature-icon" />
          </Card>
        </div>
      </section>
    </main>
  );
};

export default Content;
