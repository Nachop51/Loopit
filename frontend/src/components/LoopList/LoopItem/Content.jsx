import "./Content.css";
import PropTypes from "prop-types";
import SyntaxHighlighter from "react-syntax-highlighter";
import { atomOneDark } from "react-syntax-highlighter/dist/esm/styles/hljs";

const Content = ({ content, language }) => {
  return (
    <div className="loop-content">
      <SyntaxHighlighter
        language={language}
        style={atomOneDark}
        showLineNumbers
      >
        {content}
      </SyntaxHighlighter>
    </div>
  );
};

Content.propTypes = {
  content: PropTypes.string.isRequired,
  language: PropTypes.string,
};

export default Content;
