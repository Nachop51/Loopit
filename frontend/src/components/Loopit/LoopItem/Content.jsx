import "./Content.css";
import PropTypes from "prop-types";

const Content = ({ content, description, language }) => {
  return (
    <div className="loop-content">
      {description}
      <pre>
        <code className="code">{content}</code>
      </pre>
    </div>
  );
};

Content.propTypes = {
  content: PropTypes.string.isRequired,
  description: PropTypes.string,
};

export default Content;
