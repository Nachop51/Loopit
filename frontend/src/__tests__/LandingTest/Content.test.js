import React from "react";
import { render as testRender } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Content from "../../components/Content/index.jsx";
import { BrowserRouter as Router } from "react-router-dom";

const render = (component) => {
  testRender(<Router>{component} </Router>);
};

// TESTEO DE TODAS LAS SECCIONES DE CONTENIDO
describe("Renderizado correcto del contenedor del contenido", () => {
  it("Debe existir un componente Content en el dom", () => {
    render(<Content />);
    expect(render(<Content />)).toMatchSnapshot();
  });
});
