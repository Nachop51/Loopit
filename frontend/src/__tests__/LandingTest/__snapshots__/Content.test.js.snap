// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderizado correcto del contenedor del contenido Debe existir un componente Content en el dom 1`] = `undefined`;
